enginetest/queries/script_queries.go

```diff
@@ -2273,6 +2273,55 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "having clause without groupby clause, all rows implicitly form a single aggregate group",
+		SetUpScript: []string{
+			"create table numbers (val int);",
+			"insert into numbers values (1), (2), (3);",
+			"insert into numbers values (2), (4);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "select val from numbers;",
+				Expected: []sql.Row{{1}, {2}, {3}, {2}, {4}},
+			},
+			{
+				Query:    "select val as a from numbers having a = val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {2}, {4}},
+			},
+			{
+				Query:    "select val as a from numbers group by val having a = val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {4}},
+			},
+			{
+				Query:    "select val as a from numbers as t1 group by t1.val having a = t1.val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {4}},
+			},
+			{
+				Query:    "select t1.val as a from numbers as t1 group by 1 having a = t1.val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {4}},
+			},
+			{
+				Query:    "select t1.val as a from numbers as t1 having a = t1.val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {2}, {4}},
+			},
+			{
+				Query:    "select count(*) from numbers having count(*) = 5;",
+				Expected: []sql.Row{{5}},
+			},
+			{
+				// MySQL returns `Unknown column 'val' in 'having clause'` error for this query,
+				// but GMS builds GroupBy for any aggregate function.
+				SkipResultsCheck: true,
```

Comment:
(minor) It would be *slightly* preferable to use the new `Skip` property in `ScriptTestAssertion` (very new! I just added that a week or two ago.)  That way, we get a recorded skipped test in the test output and one day we can audit those or at least keep track of that increasing number.